### PR TITLE
Multi port

### PR DESCRIPTION
**Pull request by olbjo**
This PR is marked draft since I need to check that updates aligns with required updates in pnet struct before its merged.

### CODE REVIEW

File: src/ports/linux/sampleapp_main.c [hunk 1]
```diff
@@ -365,19 +365,19 @@ int main (int argc, char * argv[])
 
    /* Prepare stack config with IP address, gateway, station name etc */
    app_adjust_stack_configuration (&pnet_default_cfg);
-   app_copy_ip_to_struct (&pnet_default_cfg.ip_addr, ip);
-   app_copy_ip_to_struct (&pnet_default_cfg.ip_gateway, gateway);
-   app_copy_ip_to_struct (&pnet_default_cfg.ip_mask, netmask);
+   app_copy_ip_to_struct (&pnet_default_cfg.if_cfg.ip_cfg.ip_addr, ip);
```
Comment (pyhys) [inline src/ports/linux/sampleapp_main.c:368]: Maybe I missed it, but don't we need to put in `arguments.eth_interface` into two places in the config now that it not is part of `pnet_init()`?

**Review by pyhys — COMMENTED** [review]
Great!
Just a few minor changes needed.

**Comment by pyhys** [comment]
Remember to remove `interface_name` from `struct pnet`

File: doc/multiple_ports.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            |    br0      |
+            +------+------+
+            | eth0 | eth1 |
+            +------+------+
+
+
+p-net supports multiple ethernet ports. To use multiple ports the ports
+shall be grouped into a bridge. In such a configuration the profinet 
+interface is the bridge and the profinet ports are the ethernet interfaces.
+
```
Comment (fredrikdanielmoller) [inline doc/multiple_ports.rst:20]: Add a paragraph or section about terminology. In particular, the word "interface" is a cause of confusion as it has a different meaning in Profinet. In Profinet terminology, br0, eth0 and eth1 will together form an "interface". br0 will be the "management port" while eth0 and eth1 will be "physical ports".

File: include/pnet_api.h [hunk 3]
```diff
@@ -1172,23 +1172,51 @@ typedef struct pnet_ethaddr
...
  */
-typedef struct pnet_lldp_port_cfg
+typedef struct pnet_netif
 {
+   char if_name[PNET_INTERFACE_NAME_MAX_SIZE]; /**< Terminated string */
+   pnet_ethaddr_t eth_addr;                    /**< Interface MAC address */
+} pnet_netif_t;
+
+/**
+ * Port configuration including network interface and
```
Comment (fredrikdanielmoller) [inline include/pnet_api.h:1184]: Suggested description: "Physical port configuration".

File: include/pnet_api.h [hunk 4]
```diff
@@ -1172,23 +1172,51 @@ typedef struct pnet_ethaddr
...
 {
-   pnet_lldp_port_cfg_t ports[PNET_MAX_PORT];
-} pnet_lldp_cfg_t;
+   bool dhcp_enable; /**< Not supported by stack. */
+   pnet_cfg_ip_addr_t ip_addr;
+   pnet_cfg_ip_addr_t ip_mask;
+   pnet_cfg_ip_addr_t ip_gateway;
+} pnet_ip_cfg_t;
+
+/**
```
Comment (fredrikdanielmoller) [inline include/pnet_api.h:1206]: Perhaps use Profinet vocabulary instead, where "interface" is an abstract group of ports?

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: doc/multiple_ports.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+To create a bridge and add interfaces to it create the following file and content::
+
+# /etc/systemd/network/bridge-br0.netdev
```
[the remark this answers is not in the document]
Comment (olbjo) [reply doc/multiple_ports.rst:23]: Done

File: doc/multiple_ports.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
+        inet 127.0.0.1  netmask 255.0.0.0
+        inet6 ::1  prefixlen 128  scopeid 0x10<host>
+        loop  txqueuelen 1000  (Local Loopback)
+        RX packets 0  bytes 0 (0.0 B)
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 0  bytes 0 (0.0 B)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+
+Configuration of bridge using linux
```
[the remark this answers is not in the document]
Comment (olbjo) [reply doc/multiple_ports.rst:79]: Done

File: sample_app/sampleapp_common.c [hunk 7]
```diff
@@ -1284,6 +1284,7 @@ void app_plug_dap (pnet_t * net, void * arg)
 int app_adjust_stack_configuration (pnet_cfg_t * stack_config)
 {
    memset (stack_config, 0, sizeof (pnet_cfg_t));
+   stack_config->tick_us = TICK_INTERVAL_US;
```
[the remark this answers is not in the document]
Comment (olbjo) [reply sample_app/sampleapp_common.c:1287]: Done

File: src/common/pf_scheduler.c [hunk 8]
```diff
@@ -222,7 +222,8 @@ void pf_scheduler_init (pnet_t * net, uint32_t tick_interval)
    }
    memset ((void *)net->scheduler_timeouts, 0, sizeof (net->scheduler_timeouts));
 
-   net->scheduler_tick_interval = tick_interval; /* Cannot be zero */
+   net->scheduler_tick_interval = tick_interval;
+   CC_ASSERT (net->scheduler_tick_interval > 0);
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/common/pf_scheduler.c:226]: Done

File: src/device/pf_block_writer.c [hunk 9]
```diff
@@ -3855,8 +3855,8 @@ void pf_put_pdinterface_data_real (
 
    /* Owner MAC Address  */
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_block_writer.c:3856]: Done

File: src/ports/linux/sampleapp_main.c (hunk 1 above)
Comment (olbjo) [reply src/ports/linux/sampleapp_main.c:368]: You are correct. A bit worrying that I did miss this and the tests I ran passed. 
Done

File: include/pnet_api.h (hunk 4 above)
Comment (olbjo) [reply include/pnet_api.h:1206]: I tried some renaming and rewording. Hopefully to the better.

File: include/pnet_api.h (hunk 3 above)
Comment (olbjo) [reply include/pnet_api.h:1184]: Done

File: doc/multiple_ports.rst (hunk 2 above)
Comment (olbjo) [reply doc/multiple_ports.rst:20]: I have tried to clarify this.

File: doc/multiple_ports.rst [hunk 10]
```diff
@@ -0,0 +1,88 @@
+Multiple ports
+==============
+
+
+Configuration of bridge using linux
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/multiple_ports.rst]: This is now done

File: doc/multiple_ports.rst [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 0  bytes 0 (0.0 B)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+
+Configuration of bridge using linux
+-----------------------------------
+To run p-net and the sample application with multiple ports a couple
+of things needs to be done::
+
+* Reconfigure setting ``PNET_MAX_PORT`` to the actual number of ports available in the system. Rebuild.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/multiple_ports.rst]: This is now done

**Review by pyhys — APPROVED** [review]
Great work!

File: src/pf_types.h [hunk 12]
```diff
@@ -2698,10 +2698,6 @@ typedef struct pf_lldp_port
 /**
  * Port runtime data
  * Note that pnet_lldp_cfg_t  (MAC address etc) is located in the config
```
Comment (fredrikdanielmoller) [inline src/pf_types.h:2700]: Update comment

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: doc/multiple_ports.rst [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+Interface
+    Abstract group of ports. This is a cause of confusion as it has this special meaning 
+    in Profinet. Note that in profinet context, interface typically doesn't means a 
+    specific network interface. 
+Port
+    Network interface. The physical connectors are referred to as "physical ports". 
+    A "managed port" is the network interfaces to which a controller/ PLC connects.
+
+In the example described in this section, br0 is the managed port
```
[the remark this answers is not in the document]
Comment (olbjo) [reply doc/multiple_ports.rst:19]: Done

File: doc/multiple_ports.rst [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+            +------+------+
+            | eth0 | eth1 |
+            +------+------+
+
+
+p-net supports multiple ethernet ports. To use multiple ports the these
+shall be grouped into a bridge. In such a configuration the managed port / main network interface 
+is the bridge and the ethernet interfaces are named physical ports.
+
+To create a bridge and add interfaces to it, create the following files and content::
```
[the remark this answers is not in the document]
Comment (olbjo) [reply doc/multiple_ports.rst:39]: Done

File: doc/multiple_ports.rst [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+Terminology
+-----------
+
+Interface
+    Abstract group of ports. This is a cause of confusion as it has this special meaning 
+    in Profinet. Note that in profinet context, interface typically doesn't means a 
+    specific network interface. 
+Port
+    Network interface. The physical connectors are referred to as "physical ports". 
+    A "managed port" is the network interfaces to which a controller/ PLC connects.
```
[the remark this answers is not in the document]
Comment (olbjo) [reply doc/multiple_ports.rst]: Done

File: src/pf_types.h (hunk 12 above)
Comment (olbjo) [reply src/pf_types.h]: Done

**Comment by olbjo** [comment]
Thanks @fredrikdanielmoller for helping me out with the documentation. I have merged all your suggestions.

**Review by fredrikdanielmoller — APPROVED** [review]

**Review by hefloryd — APPROVED** [review]

File: doc/multiple_ports.rst [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+        loop  txqueuelen 1000  (Local Loopback)
+        RX packets 0  bytes 0 (0.0 B)
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 0  bytes 0 (0.0 B)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+
+Configuration of p-net stack and sample application (TBD)
+---------------------------------------------------------
+To run p-net and the sample application with multiple ports a couple
+of things needs to be done:
```
[the remark this answers is not in the document]
Comment (olbjo) [reply doc/multiple_ports.rst]: Done

File: doc/multiple_ports.rst [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+Terminology
+-----------
+
+Interface
+    Abstract group of ports. This is a cause of confusion as it has this special meaning 
+    in Profinet. Note that in profinet context, interface typically doesn't means a 
+    specific network interface. 
+Port
+    Network interface. The physical connectors are referred to as "physical ports". 
+    A "management port" is the network interfaces to which a controller/ PLC connects.
```
[the remark this answers is not in the document]
Comment (olbjo) [reply doc/multiple_ports.rst]: Done

File: doc/multiple_ports.rst [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+
+This section describes how to configure the p-net stack, sample application
+and system for multiple network interfaces or ports.
+
+
+Terminology
+-----------
+
+Interface
+    Abstract group of ports. This is a cause of confusion as it has this special meaning
```
[the remark this answers is not in the document]
Comment (olbjo) [reply doc/multiple_ports.rst]: Done

**Pull request merged**